tests/baselines/reference/unicodeEscapesInNames01(target=es5).js

```diff
@@ -105,6 +121,30 @@ export class PrivateIdentifierWithExtendedEscape2 {
 }
 
 
+//// [identifierVariableWithEscape1.js]
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.x = void 0;
+exports.\u0078 = 10;

```

Comment:
I thought we were expecting this to be `exports.x`

Reply:
😫

Reply:
See @rbuckton's comment.

Reply:
No, I don't think we do. There is a caveat that astral characters can't be represented with a surrogate pair of regular escapes (i.e. `\uXXXX\uXXXX`), but in theory we should just emit those as regular identifiers - plus we refuse to even parse any identifier character in the astral plane in `ES3` and `ES5` so it's a non-issue.

See the test `unicodeEscapesInNames02.ts` (specifically https://github.com/microsoft/TypeScript/blob/bdedd915dd4bb68e26ff0bf1d6659179f3e9ea54/tests/cases/compiler/unicodeEscapesInNames02.ts#L34) for an example which should be an error in ES2015 as well.

Reply:
Why do lines 127 and 128 have different levels of escaping?

Reply:
We preserve the original identifier when we can. Non-extended escapes (i.e. `\uXXXX`) are always supported and can be preserved.

Extended escapes are only supported in ES2015, and can only be preserved when targeting ES2015 and later.

Reply:
Ah, I reread - I guess we lose the original node contents in the module transform, but that behavior has always been the case. Maybe we aren't setting the original node?

Reply:
Gross, but not a regression?

Reply:
[No, that's the same behavior as before.](https://www.typescriptlang.org/play?target=2&module=1#code/KYDwDg9gTgLgBAG2PAOgVwAwYOwA44C8cAjBgNwCwAUCANS1lA)